esxi/tests/test_unit.py

```diff
@@ -1210,3 +1214,37 @@ def test_use_configured_hostname(vcsim_instance, dd_run_check, aggregator, datad
             ]
         },
     )
+
+
+@pytest.mark.usefixtures("service_instance")
+def test_use_non_socks_proxy(vcsim_instance, dd_run_check, caplog):
+    instance = copy.deepcopy(vcsim_instance)
+    instance['proxy'] = "http://localhost"
+    caplog.set_level(logging.WARNING)
+    check = EsxiCheck('esxi', {}, [instance])
+    dd_run_check(check)
+    assert "Proxy scheme http not supported; ignoring" in caplog.text
+
+
+@pytest.mark.usefixtures("service_instance")
+def test_use_socks_proxy(vcsim_instance, dd_run_check, caplog, aggregator):
+    instance = copy.deepcopy(vcsim_instance)
+    instance['proxy'] = "socks5://test"
```

Comment:
This proxy value is not causing an error or exception when calling `socks.create_connection`?

Reply:
It is not because we are using the mock service instance